Form/ChoiceList/PhpcrOdmQueryBuilderLoader.php

```diff
@@ -77,10 +80,16 @@ public function getEntities()
      */
     public function getEntitiesByIds($identifier, array $values)
     {
+        /* performance: if we could figure out whether the query builder is "
+         * empty" (that is only checking for the class) we could optimize this
+         * to a $this->dm->findMany(null, $values)
+         */
+
         $qb = clone $this->queryBuilder;
         $alias = $qb->getPrimaryAlias();
+        $where = $qb->andWhere()->orX();
         foreach ($values as $val) {
-            $qb->orWhere()->eq()->field($alias.'.'.$identifier)->literal($val);
+            $where->same($val, $alias);
```

Comment:
Ups, i think you are right dan. @lsmith77 can you maybe add a second document to the fixtures and test with one and two ids? Right now we dont test that we actually filter vs just loading all.